File: app/src/main/java/com/example/myapplication/feature/catalog/ExerciseCatalogScreen.kt
package com.example.myapplication.feature.catalog

import androidx.compose.animation.animateContentSize
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.myapplication.core.model.Equipment
import com.example.myapplication.core.model.ExerciseDefinition
import com.example.myapplication.core.model.MuscleGroup
import com.example.myapplication.ui.theme.EnergyOrange
import com.example.myapplication.ui.theme.SuccessGreen
import com.example.myapplication.ui.theme.customColors

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ExerciseCatalogScreen(
    exercises: List<ExerciseDefinition>,
    onBack: (() -> Unit)? = null,
) {
    val colors = MaterialTheme.colorScheme
    val customColors = colors.customColors

    var searchQuery by remember { mutableStateOf("") }
    var selectedMuscle by remember { mutableStateOf<MuscleGroup?>(null) }
    var selectedEquipment by remember { mutableStateOf<Equipment?>(null) }

    val filteredExercises = remember(searchQuery, selectedMuscle, selectedEquipment, exercises) {
        exercises.filter { exercise ->
            val matchesSearch = exercise.nameVi.contains(searchQuery, ignoreCase = true) ||
                    exercise.instructionsVi.any { it.contains(searchQuery, ignoreCase = true) }
            val matchesMuscle = selectedMuscle == null || exercise.primaryMuscle == selectedMuscle || exercise.secondaryMuscles.contains(selectedMuscle)
            val matchesEquipment = selectedEquipment == null || exercise.equipment.contains(selectedEquipment)
            matchesSearch && matchesMuscle && matchesEquipment
        }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Thư viện bài tập 📚", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    if (onBack != null) {
                        TextButton(onClick = onBack) {
                            Text("◀", color = EnergyOrange, fontSize = 20.sp)
                        }
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = colors.background,
                    titleContentColor = customColors.primaryText,
                    navigationIconContentColor = customColors.primaryText
                )
            )
        },
        containerColor = colors.background
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
        ) {
            // Search Bar
            OutlinedTextField(
                value = searchQuery,
                onValueChange = { searchQuery = it },
                placeholder = { Text("Tìm kiếm tên bài tập, hướng dẫn...") },
                singleLine = true,
                shape = RoundedCornerShape(14.dp),
                colors = OutlinedTextFieldDefaults.colors(
                    focusedBorderColor = EnergyOrange,
                    unfocusedBorderColor = colors.outline
                ),
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 8.dp)
            )

            // Muscle filters
            Text(
                "Nhóm cơ tập trung",
                style = MaterialTheme.typography.titleSmall,
                fontWeight = FontWeight.Bold,
                color = customColors.primaryText,
                modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp)
            )
            LazyRow(
                contentPadding = PaddingValues(horizontal = 20.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp)
            ) {
                item {
                    FilterChip(
                        selected = selectedMuscle == null,
                        onClick = { selectedMuscle = null },
                        label = { Text("Tất cả") }
                    )
                }
                items(MuscleGroup.entries) { muscle ->
                    FilterChip(
                        selected = selectedMuscle == muscle,
                        onClick = { selectedMuscle = muscle },
                        label = { Text(muscleLabelVi(muscle)) }
                    )
                }
            }

            // Equipment filters
            Text(
                "Dụng cụ cần thiết",
                style = MaterialTheme.typography.titleSmall,
                fontWeight = FontWeight.Bold,
                color = customColors.primaryText,
                modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp)
            )
            LazyRow(
                contentPadding = PaddingValues(horizontal = 20.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp)
            ) {
                item {
                    FilterChip(
                        selected = selectedEquipment == null,
                        onClick = { selectedEquipment = null },
                        label = { Text("Tất cả") }
                    )
                }
                items(Equipment.entries) { equipment ->
                    FilterChip(
                        selected = selectedEquipment == equipment,
                        onClick = { selectedEquipment = equipment },
                        label = { Text(equipmentLabelVi(equipment)) }
                    )
                }
            }

            // Results List
            LazyColumn(
                contentPadding = PaddingValues(horizontal = 20.dp, vertical = 12.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp),
                modifier = Modifier.weight(1f).fillMaxWidth()
            ) {
                if (filteredExercises.isEmpty()) {
                    item {
                        Box(
                            modifier = Modifier.fillParentMaxSize(),
                            contentAlignment = Alignment.Center
                        ) {
                            Text("Không tìm thấy bài tập nào phù hợp 🔍", color = customColors.mutedText)
                        }
                    }
                } else {
                    items(filteredExercises, key = { it.id }) { exercise ->
                        CatalogExerciseCard(exercise)
                    }
                }
            }
        }
    }
}

@Composable
private fun CatalogExerciseCard(exercise: ExerciseDefinition) {
    val colors = MaterialTheme.colorScheme
    val customColors = colors.customColors
    var expanded by remember { mutableStateOf(false) }

    Column(
        Modifier
            .fillMaxWidth()
            .animateContentSize()
            .background(colors.surfaceVariant, RoundedCornerShape(16.dp))
            .border(1.dp, colors.outline, RoundedCornerShape(16.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Box(
                modifier = Modifier
                    .size(36.dp)
                    .clip(CircleShape)
                    .background(EnergyOrange),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    muscleEmoji(exercise.primaryMuscle),
                    fontSize = 18.sp
                )
            }
            Spacer(Modifier.width(12.dp))

            Column(Modifier.weight(1f)) {
                Text(
                    exercise.nameVi,
                    style = MaterialTheme.typography.titleMedium,
                    color = customColors.primaryText,
                    fontWeight = FontWeight.SemiBold
                )
                Text(
                    "${muscleLabelVi(exercise.primaryMuscle)} · ${exercise.equipment.joinToString { equipmentLabelVi(it) }}",
                    color = customColors.mutedText,
                    style = MaterialTheme.typography.bodySmall
                )
            }
        }

        // Expandable instructions
        Text(
            if (expanded) "Ẩn hướng dẫn ▲" else "Xem hướng dẫn ▼",
            color = EnergyOrange,
            style = MaterialTheme.typography.labelMedium,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier
                .fillMaxWidth()
                .clickable { expanded = !expanded }
                .padding(vertical = 4.dp),
        )

        if (expanded) {
            Column(
                Modifier
                    .fillMaxWidth()
                    .background(colors.surface, RoundedCornerShape(12.dp))
                    .padding(14.dp),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                exercise.instructionsVi.forEachIndexed { index, instruction ->
                    Row {
                        Text(
                            "${index + 1}.",
                            color = EnergyOrange,
                            fontWeight = FontWeight.Bold,
                            modifier = Modifier.width(24.dp)
                        )
                        Text(instruction, color = customColors.primaryText, style = MaterialTheme.typography.bodySmall)
                    }
                }
            }
        }
    }
}

private fun muscleLabelVi(muscle: MuscleGroup): String = when (muscle) {
    MuscleGroup.CHEST -> "Ngực"
    MuscleGroup.BACK -> "Lưng"
    MuscleGroup.SHOULDERS -> "Vai"
    MuscleGroup.BICEPS -> "Tay trước"
    MuscleGroup.TRICEPS -> "Tay sau"
    MuscleGroup.CORE -> "Bụng"
    MuscleGroup.QUADS -> "Đùi trước"
    MuscleGroup.HAMSTRINGS -> "Đùi sau"
    MuscleGroup.GLUTES -> "Mông"
    MuscleGroup.CALVES -> "Bắp chân"
    MuscleGroup.FULL_BODY -> "Toàn thân"
    MuscleGroup.CARDIO -> "Tim mạch"
    MuscleGroup.MOBILITY -> "Di động"
}

private fun equipmentLabelVi(eq: Equipment): String = when (eq) {
    Equipment.BODYWEIGHT -> "Không dụng cụ"
    Equipment.DUMBBELL -> "Tạ đơn"
    Equipment.BAND -> "Dây kháng lực"
    Equipment.BARBELL -> "Tạ đòn"
    Equipment.BENCH -> "Ghế băng"
    Equipment.CABLE -> "Cáp"
    Equipment.MACHINE -> "Máy tập"
    Equipment.CARDIO_MACHINE -> "Máy chạy/đạp xe"
}

private fun muscleEmoji(muscle: MuscleGroup): String = when (muscle) {
    MuscleGroup.CHEST -> "🫁"
    MuscleGroup.BACK -> "🔙"
    MuscleGroup.SHOULDERS -> "💪"
    MuscleGroup.BICEPS -> "💪"
    MuscleGroup.TRICEPS -> "💪"
    MuscleGroup.CORE -> "🎯"
    MuscleGroup.QUADS -> "🦵"
    MuscleGroup.HAMSTRINGS -> "🦵"
    MuscleGroup.GLUTES -> "🍑"
    MuscleGroup.CALVES -> "🦶"
    MuscleGroup.FULL_BODY -> "🏋️"
    MuscleGroup.CARDIO -> "❤️"
    MuscleGroup.MOBILITY -> "🧘"
}

File: flutter/lib/feature/catalog/exercise_catalog_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../core/model/catalog_models.dart';
import '../../data/providers/data_providers.dart';
import '../../ui/theme/colors.dart';
import '../../ui/theme/theme.dart';
import '../../ui/components/exercise_3d_dialog.dart';
import '../../ui/theme/radius.dart';
import '../../ui/theme/typography.dart';
import '../../ui/components/gym_card.dart';

extension EquipmentLabelVi on Equipment {
  String labelVi() {
    switch (this) {
      case Equipment.bodyweight:
        return "Không dụng cụ";
      case Equipment.dumbbell:
        return "Tạ đơn";
      case Equipment.band:
        return "Dây kháng lực";
      case Equipment.barbell:
        return "Tạ đòn";
      case Equipment.bench:
        return "Ghế băng";
      case Equipment.cable:
        return "Cáp";
      case Equipment.machine:
        return "Máy tập";
      case Equipment.cardioMachine:
        return "Máy chạy/đạp xe";
    }
  }
}

extension MuscleGroupEmoji on MuscleGroup {
  String emoji() {
    switch (this) {
      case MuscleGroup.chest:
        return "🫁";
      case MuscleGroup.back:
        return "🔙";
      case MuscleGroup.shoulders:
      case MuscleGroup.biceps:
      case MuscleGroup.triceps:
        return "💪";
      case MuscleGroup.core:
        return "🎯";
      case MuscleGroup.quads:
      case MuscleGroup.hamstrings:
        return "🦵";
      case MuscleGroup.glutes:
        return "🍑";
      case MuscleGroup.calves:
        return "🦶";
      case MuscleGroup.fullBody:
        return "🏋️";
      case MuscleGroup.cardio:
        return "❤️";
      case MuscleGroup.mobility:
        return "🧘";
    }
  }
}

class ExerciseCatalogScreen extends ConsumerStatefulWidget {
  final VoidCallback onBack;

  const ExerciseCatalogScreen({super.key, required this.onBack});

  @override
  ConsumerState<ExerciseCatalogScreen> createState() =>
      _ExerciseCatalogScreenState();
}

class _ExerciseCatalogScreenState extends ConsumerState<ExerciseCatalogScreen> {
  final TextEditingController _searchController = TextEditingController();
  String _searchQuery = "";
  MuscleGroup? _selectedMuscle;
  Equipment? _selectedEquipment;

  @override
  void initState() {
    super.initState();
    _searchController.addListener(() {
      setState(() {
        _searchQuery = _searchController.text;
      });
    });
  }

  @override
  void dispose() {
    _searchController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final catalogRepo = ref.watch(assetCatalogRepositoryProvider);
    final exercises = catalogRepo.exercises;
    final isDark = Theme.of(context).brightness == Brightness.dark;
    final customColors = context.customColors;

    final filteredExercises = exercises.where((exercise) {
      final matchesSearch =
          exercise.nameVi.toLowerCase().contains(_searchQuery.toLowerCase()) ||
          exercise.instructionsVi.any(
            (ins) => ins.toLowerCase().contains(_searchQuery.toLowerCase()),
          );
      final matchesMuscle =
          _selectedMuscle == null ||
          exercise.primaryMuscleGroup == _selectedMuscle ||
          exercise.secondaryMuscleGroups.contains(_selectedMuscle);
      final matchesEquipment =
          _selectedEquipment == null ||
          exercise.equipment.contains(_selectedEquipment);
      return matchesSearch && matchesMuscle && matchesEquipment;
    }).toList();

    return Scaffold(
      backgroundColor: isDark ? AppColors.darkBg : AppColors.white,
      appBar: AppBar(
        title: Text(
          "Thư viện bài tập 📚",
          style: TextStyle(
            color: customColors.primaryText,
            fontWeight: FontWeight.bold,
            fontSize: 18,
          ),
        ),
        leading: IconButton(
          key: const ValueKey("catalog-back-button"),
          icon: const Icon(Icons.arrow_back),
          color: customColors.primaryText,
          onPressed: widget.onBack,
        ),
        backgroundColor: isDark ? AppColors.darkBg : AppColors.white,
        elevation: 0,
        scrolledUnderElevation: 0,
      ),
      body: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // Search Bar
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 8),
              child: TextField(
                controller: _searchController,
                decoration: InputDecoration(
                  hintText: "Tìm kiếm tên bài tập, hướng dẫn...",
                  hintStyle: TextStyle(
                    color: customColors.mutedText,
                    fontSize: 14,
                  ),
                  prefixIcon: Icon(Icons.search, color: customColors.mutedText),
                  filled: true,
                  fillColor: isDark
                      ? AppColors.darkSurface
                      : AppColors.surfaceGray,
                  contentPadding: const EdgeInsets.symmetric(
                    vertical: 0,
                    horizontal: 16,
                  ),
                  border: OutlineInputBorder(
                    borderRadius: BorderRadius.circular(14),
                    borderSide: BorderSide.none,
                  ),
                  focusedBorder: OutlineInputBorder(
                    borderRadius: BorderRadius.circular(14),
                    borderSide: const BorderSide(
                      color: AppColors.energyOrange,
                      width: 1.5,
                    ),
                  ),
                ),
                style: TextStyle(color: customColors.primaryText, fontSize: 14),
              ),
            ),

            // Muscle filters
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 4),
              child: Text(
                "Nhóm cơ tập trung",
                style: TextStyle(
                  color: customColors.primaryText,
                  fontWeight: FontWeight.bold,
                  fontSize: 14,
                ),
              ),
            ),
            SizedBox(
              height: 48,
              child: ListView(
                scrollDirection: Axis.horizontal,
                padding: const EdgeInsets.symmetric(horizontal: 16),
                children: [
                  _buildFilterChip(
                    label: "Tất cả",
                    selected: _selectedMuscle == null,
                    onSelected: (_) {
                      setState(() {
                        _selectedMuscle = null;
                      });
                    },
                    isDark: isDark,
                  ),
                  ...MuscleGroup.values.map((muscle) {
                    return _buildFilterChip(
                      label: muscle.labelVi(),
                      selected: _selectedMuscle == muscle,
                      onSelected: (_) {
                        setState(() {
                          _selectedMuscle = muscle;
                        });
                      },
                      isDark: isDark,
                    );
                  }),
                ],
              ),
            ),

            // Equipment filters
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 4),
              child: Text(
                "Dụng cụ cần thiết",
                style: TextStyle(
                  color: customColors.primaryText,
                  fontWeight: FontWeight.bold,
                  fontSize: 14,
                ),
              ),
            ),
            SizedBox(
              height: 48,
              child: ListView(
                scrollDirection: Axis.horizontal,
                padding: const EdgeInsets.symmetric(horizontal: 16),
                children: [
                  _buildFilterChip(
                    label: "Tất cả",
                    selected: _selectedEquipment == null,
                    onSelected: (_) {
                      setState(() {
                        _selectedEquipment = null;
                      });
                    },
                    isDark: isDark,
                  ),
                  ...Equipment.values.map((eq) {
                    return _buildFilterChip(
                      label: eq.labelVi(),
                      selected: _selectedEquipment == eq,
                      onSelected: (_) {
                        setState(() {
                          _selectedEquipment = eq;
                        });
                      },
                      isDark: isDark,
                    );
                  }),
                ],
              ),
            ),

            const SizedBox(height: 8),

            // Results List
            Expanded(
              child: filteredExercises.isEmpty
                  ? Center(
                      child: Text(
                        "Không tìm thấy bài tập nào phù hợp 🔍",
                        style: TextStyle(
                          color: customColors.mutedText,
                          fontSize: 14,
                        ),
                      ),
                    )
                  : ListView.builder(
                      padding: const EdgeInsets.symmetric(
                        horizontal: 20,
                        vertical: 12,
                      ),
                      itemCount: filteredExercises.length,
                      itemBuilder: (context, index) {
                        return Padding(
                          padding: const EdgeInsets.only(bottom: 12),
                          child: _CatalogExerciseCard(
                            exercise: filteredExercises[index],
                          ),
                        );
                      },
                    ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildFilterChip({
    required String label,
    required bool selected,
    required ValueChanged<bool> onSelected,
    required bool isDark,
  }) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 4, vertical: 6),
      child: ChoiceChip(
        label: Text(
          label,
          style: TextStyle(
            color: selected
                ? Colors.white
                : (isDark ? AppColors.darkText : AppColors.navy),
            fontSize: 12,
            fontWeight: selected ? FontWeight.bold : FontWeight.normal,
          ),
        ),
        selected: selected,
        onSelected: onSelected,
        selectedColor: AppColors.energyOrange,
        backgroundColor: isDark ? AppColors.darkSurface : AppColors.surfaceGray,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
        side: BorderSide.none,
        showCheckmark: false,
      ),
    );
  }
}

class _CatalogExerciseCard extends StatefulWidget {
  final ExerciseDefinition exercise;

  const _CatalogExerciseCard({required this.exercise});

  @override
  State<_CatalogExerciseCard> createState() => _CatalogExerciseCardState();
}

class _CatalogExerciseCardState extends State<_CatalogExerciseCard> {
  bool _expanded = false;

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;
    final customColors = context.customColors;

    return GymCard(
      variant: GymCardVariant.outlined,
      padding: const EdgeInsets.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Container(
                width: 36,
                height: 36,
                decoration: const BoxDecoration(
                  color: AppColors.energyOrange,
                  shape: BoxShape.circle,
                ),
                alignment: Alignment.center,
                child: Text(
                  widget.exercise.primaryMuscleGroup.emoji(),
                  style: const TextStyle(fontSize: 18),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      widget.exercise.nameVi,
                      style: GymTypography.titleMedium.bold.copyWith(
                        color: customColors.primaryText,
                      ),
                    ),
                    const SizedBox(height: 2),
                    Text(
                      "${widget.exercise.primaryMuscleGroup.labelVi()} · ${widget.exercise.equipment.map((e) => e.labelVi()).join(', ')}",
                      style: isDark
                          ? GymTypography.bodySmall.mutedDark
                          : GymTypography.bodySmall.muted,
                    ),
                  ],
                ),
              ),
            ],
          ),

          const SizedBox(height: 10),

          // Expand/Collapse label
          InkWell(
            onTap: () {
              setState(() {
                _expanded = !_expanded;
              });
            },
            child: Padding(
              padding: const EdgeInsets.symmetric(vertical: 4),
              child: Row(
                children: [
                  Text(
                    _expanded ? "Ẩn hướng dẫn ▲" : "Xem hướng dẫn ▼",
                    style: GymTypography.labelSmall.orange.bold,
                  ),
                ],
              ),
            ),
          ),

          if (_expanded) ...[
            const SizedBox(height: 8),
            Container(
              width: double.infinity,
              padding: const EdgeInsets.all(14),
              decoration: BoxDecoration(
                color: isDark
                    ? AppColors.darkSurfaceVariant
                    : AppColors.surfaceGray,
                borderRadius: GymRadius.mdBorder,
              ),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  ...widget.exercise.instructionsVi.asMap().entries.map((
                    entry,
                  ) {
                    final index = entry.key;
                    final instruction = entry.value;
                    return Padding(
                      padding: const EdgeInsets.only(bottom: 6),
                      child: Row(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text(
                            "${index + 1}.",
                            style: GymTypography.bodyMedium.orange.bold,
                          ),
                          const SizedBox(width: 8),
                          Expanded(
                            child: Text(
                              instruction,
                              style: GymTypography.bodyMedium.copyWith(
                                color: customColors.primaryText,
                              ),
                            ),
                          ),
                        ],
                      ),
                    );
                  }),
                  if (widget.exercise.animationId != null) ...[
                    const SizedBox(height: 10),
                    SizedBox(
                      width: double.infinity,
                      height: 48,
                      child: ElevatedButton(
                        key: ValueKey("exercise-3d-btn-${widget.exercise.id}"),
                        onPressed: () {
                          Exercise3DDialog.show(
                            context: context,
                            animationId: widget.exercise.animationId!,
                            exerciseName: widget.exercise.nameVi,
                            instructions: widget.exercise.instructionsVi,
                          );
                        },
                        style: ElevatedButton.styleFrom(
                          backgroundColor: AppColors.energyOrange,
                          foregroundColor: Colors.white,
                          shape: RoundedRectangleBorder(
                            borderRadius: BorderRadius.circular(10),
                          ),
                          elevation: 0,
                          padding: EdgeInsets.zero,
                        ),
                        child: Text(
                          "Xem chuyển động 3D",
                          style: GymTypography.titleSmall.white.bold,
                        ),
                      ),
                    ),
                  ],
                ],
              ),
            ),
          ],
        ],
      ),
    );
  }
}
